### DISCUSSION TITLE
Exact matches in matchCurrentValue, matchNewValue

### DISCUSSION THREAD

**Discussion by amezin** [renovatebot/renovate#29167] [Suggest an Idea]
### Tell us more.

I was surprised that `matchCurrentValue` and `matchNewValue` in `packageRules` don't take simple strings.

One problem is that

    DEBUG: matchNewValue should be a regex, starting and ending with `/`

Is a `DEBUG` level message - not visible on dependency dashboard. While it is clearly a configuration error, if only regex patterns are accepted.

Another problem is that instead of writing `"6.0.0"` you have to write `"/6\\.0\\.0/"`, escaping the value twice - first from literal string to regex, then double-escaping backslashes for JSON.

Why aren't globs or, at least, exact string matches allowed?

**Comment by rarkins** [comment]
If you want to improve the config validation, you can contribute it here: https://github.com/renovatebot/renovate/tree/main/lib/config

We could also consider converting these to our "regex or glob" matching, which would allow exact matches as exact matches are value glob. PRs also welcome for that

**Reply by amezin** [reply]
By the way, `matchNewVersion` would likely match my use case a bit better. I see that there's `matchCurrentVersion`, but no `matchNewVersion`. Are there any reasons why it's not implemented - besides "nobody needed it before"?

**Reply by rarkins** [reply]
I think you guessed right
